### ISSUE TITLE
Individual border properties in radio and checkbox are being merged into border-shorthand syntax during webpack build with production flag breaking PostCSS custom properties transform 

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#1199]
# Issue Description
When webpack is run in production mode, some of the checkbox and radio button theme styles are converted into a shorthand syntax which breaks the PostCSS plugin transformation of CSS custom properties to static values. This issue was originally discovered in the cerner-consumer-theme repo: https://github.cerner.com/terra/cerner-consumer-theme/issues/28

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
Radio and checkbox styles can be custom themed

## Current Behavior
Individual border properties in radio and checkbox are being merged into border-shorthand syntax when webpack is run in production mode which causes issues when trying to compile static values via the PostCSS custom properties plugin.

## Extended description
Did some more debugging and came across and oddity between our styles in the local dev server and in the deployed environment.

With our local dev server running:

### Given this input

```scss
.native-input:checked + .outer-ring {
      background-color: var(--terra-form-radio-outer-ring-checked-background-color, transparent);
      border-color: var(--terra-form-radio-outer-ring-checked-border-color, transparent);
      border-radius: var(--terra-form-radio-outer-ring-checked-border-radius, 0);
      border-style: var(--terra-form-radio-outer-ring-checked-border-style, none);
      border-width: var(--terra-form-radio-outer-ring-checked-border-width, 0);
      box-sizing: border-box;
}
```

### We are getting the following output in our extracted CSS file. 
The border properties remain split out individually

```scss
[dir] .Radio__radio___1mKDF .Radio__native-input___2v9UZ:checked + .Radio__outer-ring___lGCNm {
    background-color: var(--terra-form-radio-outer-ring-checked-background-color, transparent);
    border-color: var(--terra-form-radio-outer-ring-checked-border-color, transparent);
    border-radius: var(--terra-form-radio-outer-ring-checked-border-radius, 0);
    border-style: var(--terra-form-radio-outer-ring-checked-border-style, none);
    border-width: var(--terra-form-radio-outer-ring-checked-border-width, 0);
    margin-top: var(--terra-form-radio-outer-ring-checked-margin-top, 0);
}
```

In our deployed environment for the PR deployment URLs, we run wepback with the production flag. It seems one of our style processors is merging the border rules into one property using the border shorthand.

### Given this input

```scss
.native-input:checked + .outer-ring {
      background-color: var(--terra-form-radio-outer-ring-checked-background-color, transparent);
      border-color: var(--terra-form-radio-outer-ring-checked-border-color, transparent);
      border-radius: var(--terra-form-radio-outer-ring-checked-border-radius, 0);
      border-style: var(--terra-form-radio-outer-ring-checked-border-style, none);
      border-width: var(--terra-form-radio-outer-ring-checked-border-width, 0);
      box-sizing: border-box;
}
```

### We are getting the following output in our extracted CSS file in the deployed code used for the PR deployment
Note all the border styles have been merged into 1 border short-hand syntax.

```scss
[dir] .Radio__radio___1mKDF .Radio__native-input___2v9UZ:checked+.Radio__outer-ring___lGCNm {
    background-color: var(--terra-form-radio-outer-ring-checked-background-color,transparent);
    border: var(--terra-form-radio-outer-ring-checked-border-width,0) var(--terra-form-radio-outer-ring-checked-border-style,none) var(--terra-form-radio-outer-ring-checked-border-color,transparent);
    border-radius: var(--terra-form-radio-outer-ring-checked-border-radius,0);
    margin-top: var(--terra-form-radio-outer-ring-checked-margin-top,0)
}
```

Not sure if postCSS or node-sass is doing this, I believe this issue is causing oddities with the theming styles mentioned above.

I've found by switching any place where all three border properties are used to the border shorthand syntax, it resolves the error where webpack is run in prod mode and the individual border styles get converted to shorthand syntax.

**Switch this**
```css
.selector {
  border-color: var(--terra-form-radio-outer-ring-checked-border-color, #f00);
  border-style: var(--terra-form-radio-outer-ring-checked-border-style, solid);
  border-width: var(--terra-form-radio-outer-ring-checked-border-width, 1px);
}
```

**to this**

```css
.selector {
  border: var(--terra-form-radio-outer-ring-checked-border, 1px solid #f00);
}
```

I have yet been able to find which of our build tools is responsible for converting this when webpack is run with the prod flag:

```css
.native-input:checked + .outer-ring {
      border-color: var(--terra-form-radio-outer-ring-checked-border-color, transparent);
      border-style: var(--terra-form-radio-outer-ring-checked-border-style, none);
      border-width: var(--terra-form-radio-outer-ring-checked-border-width, 0);
}
```

to this

```css
[dir] .Radio__radio___1mKDF .Radio__native-input___2v9UZ:checked+.Radio__outer-ring___lGCNm {
    border: var(--terra-form-radio-outer-ring-checked-border-width,0) var(--terra-form-radio-outer-ring-checked-border-style,none) var(--terra-form-radio-outer-ring-checked-border-color,transparent);

}
```

## Steps to Reproduce

1. Write border styles as individual properties with CSS custom properties, border-width, border-color, border-style. (Must include all 3 in selector, if only 1 or 2 or included, it doesn't appear to convert them to border short-hand syntax)
2. Run webpack with production flag

## Environment

* Component Version: terra-form-checkbox, terra-form-radio
* Browser Name and Version: All
* Operating System and version (desktop or mobile): All

**Comment by bjankord** [comment]
# Tech Design

I have yet been able to find which of our build tools is responsible for converting the border styles to short-hand syntax when webpack is run with the prod flag. Its likely something in node-sass or postcss. While we could figure out if there is a config we need to change to prevent that, we'd have to document that and hope consumers update their webpack configs. 

Instead, I think we should identify selectors where we are using all 3 border styles as individual properties with CSS custom properties and convert them to shorthand syntax. Essentially write our code in the way it will be transformed anyways so it becomes a non-issue.

**Switch this**

```css
.selector {
  border-color: var(--terra-form-radio-outer-ring-checked-border-color, #f00);
  border-style: var(--terra-form-radio-outer-ring-checked-border-style, solid);
  border-width: var(--terra-form-radio-outer-ring-checked-border-width, 1px);
}
```

**to this**

```css
.selector {
  border: var(--terra-form-radio-outer-ring-checked-border, 1px solid #f00);
}
```

We should try and get a stylelint rule in place to avoid this issue in the future. 
https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties/ would be a good place to start, we'll likely want make a custom rule based on that rule which targets only the border properties.

**Comment by bjankord** [comment]
Resolved in #1202 and #1217

**Comment by bjankord** [comment]
Thinking about making a stylelint rule for this. Its come up in a few other places. 
The following should detect when all 3 individual border properties are used in 1 selector. We could warn people to use shorthand so they don't run into this bug.

```jsx
import * as postcss from "postcss";

export default postcss.plugin("postcss-collect-borders", (options = {}) => {
  const borderStyles = ['border-style', 'border-color', 'border-width'];
  return root => {
    root.walkRules(rule => {
      const borderArry = [];
      rule.walkDecls(decl => {
        if (decl.prop.startsWith("border")) {
          borderArry.push(decl.prop);
        }
      });
      
      // Checks if all 3 individual border styles (width, style, color) are set
      if (borderStyles.every(style => borderArry.includes(style))) {
        console.log("This should use border shorthand instead");
      }
    });
  };
});
```
